task_processor/management/commands/runprocessor.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import signal
+import time
+import typing
+from argparse import ArgumentParser
+from datetime import timedelta
+
+from django.core.management import BaseCommand
+from django.utils import timezone
+
+from task_processor.task_registry import registered_tasks
+from task_processor.thread_monitoring import (
+    clear_unhealthy_threads,
+    write_unhealthy_threads,
+)
+from task_processor.threads import TaskRunner
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        signal.signal(signal.SIGINT, self._exit_gracefully)
+        signal.signal(signal.SIGTERM, self._exit_gracefully)
+
+        self._threads: typing.List[TaskRunner] = []
```

Comment:
Won't need the typing import if we just use `list` here and elsewhere in this file.